### PR TITLE
Support "Clone API" to filter virtualmachine.spec.template.annotation…

### LINKED ISSUE #10374: Supportr "Clone API" to filter VirtualMachine.spec.template.annotation and labels
**Is your feature request related to a problem? Please describe**:
A clear and concise description of what the problem is.
we used kube-ovn as our kubevirt cluster cni. as is known to all, this kube-ovn cni will add the network's info in "virtualmachine.spec.template.annotation"
image
it will raise an issue when we use "Clone API" to clone a new vm. the clone vm would use the same network. so we want to add annotation Filter to avoid this problems.
**Describe the solution you'd like**:

**Describe alternatives you've considered**:
```yaml
kind: VirtualMachineClone
apiVersion: "clone.kubevirt.io/v1alpha1"
metadata:
  name: testclone

spec:
  # source & target definitions
  source:
    apiGroup: kubevirt.io
    kind: VirtualMachine
    name: vm-cirros
  target:
    apiGroup: kubevirt.io
    kind: VirtualMachine
    name: vm-clone-target

  # labels & annotations definitions
  labelFilters:
  - "*"
  - "!someKey/*"
  annotationFilters:
  - "anotherKey/*"
  templateFilters:
    labelFilters:
    - "*"
    - "!someKey/*"
    annotationFilters:
    - "anotherKey/*"

  # other identity stripping specs:
  newMacAddresses:
    interfaceName: "00-11-22"
  newSMBiosSerial: "new-serial"

### PR DESCRIPTION
**Pull request by matthewei**
**What this PR does / why we need it**:
The [label--annotation-filters](https://github.com/kubevirt/user-guide/blob/main/docs/operations/clone_api.md#label--annotation-filters) strip it from the top-level just as your description and it can't strip from spec.template.metadata.annotation or labels. Some cni such as kube-ovn will inject network information into the annotation. So we need to filter them out.

please theck the following example.
```yaml
spec:
  # labels & annotations definitions
  labelFilters:
  - "*"
  - "!someKey/*"
  annotationFilters:
  - "anotherKey/*"

  template: # 
    labelFilters:
    - "*"
    - "!someKey/*"
    annotationFilters:
    - "anotherKey/*"
 ```
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10374 

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
1. Support "Clone API" to filter VirtualMachine.spec.template.annotation and VirtualMachine.spec.template.label
```

### CODE REVIEW

**Comment by matthewei** [comment]
@0xFelix hello, I make a few mistakes on the https://github.com/kubevirt/kubevirt/pull/10469 so I raise a new PR.

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go [hunk 1]
```diff
@@ -108,11 +114,12 @@ func validateFilters(filters []string, fieldName string) (causes []metav1.Status
 		return nil
 	}
 
+	errPattern := "%s filter %s is invalid: cannot contain a %s character (%s); FilterRules: %s"
```
Comment (0xFelix) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go:117]: nit: Keep definition of this at L129 and as `const`.

File: pkg/virtctl/create/clone/clone.go
Comment (0xFelix) [inline pkg/virtctl/create/clone/clone.go:1]: Can you put the changes to virtctl into a separate commit?

**Review by 0xFelix — COMMENTED** [review]
Looks OK to me but there are still failing tests. Can you please address those?

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go (hunk 1 above)
Comment (matthewei) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go:117]: get it!

File: pkg/virtctl/create/clone/clone.go
Comment (matthewei) [reply pkg/virtctl/create/clone/clone.go:1]: yes, will do it

**Comment by matthewei** [comment]
> Looks OK to me but there are still failing tests. Can you please address those?

I will check the e2e code.

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go (hunk 1 above)
Comment (matthewei) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go]: done

File: pkg/virtctl/create/clone/clone.go
Comment (matthewei) [reply pkg/virtctl/create/clone/clone.go:1]: done ,will raise another PR

**Comment by matthewei** [comment]
If this PR can be merged and I would like to modify [clone_api](https://github.com/kubevirt/user-guide/blob/main/docs/operations/clone_api.md#label--annotation-filters).

**Comment by matthewei** [comment]
@0xFelix Hi, I have finished this PR. Please move to next step. Thanks a lot.

**Comment by matthewei** [comment]
can i rais a PR for [clone_api](https://github.com/kubevirt/user-guide/blob/main/docs/operations/clone_api.md#label--annotation-filters)?

**Review by 0xFelix — COMMENTED** [review]
Thanks!

/lgtm
/retest-required

**Comment by 0xFelix** [comment]
> can I raise a PR for [clone_api](https://github.com/kubevirt/user-guide/blob/main/docs/operations/clone_api.md#label--annotation-filters)?

You mean a PR for updating the docs on the clone API? Please go ahead!
